README.md
graphite Cookbook
================
Simple Install and configure metrics monitoring tool "Graphite".

This cookbook can be used to making "Grafana" dashboard.

Requirements
------------
* CentOS6 or Amazon Linux 2014.03.1
* Chef 11 or higher
* mysql    - Opscode cookbook
* database - Opscode cookbook
* yum      - Opscode cookbook
* openssl  - Opscode cookbook

Attributes
----------
#### grahite::default
```ruby
#Graphite Web Settings
default[:graphite][:timezone]         = "Asia/Tokyo"
default[:graphite][:mysql_password]   = "graphite"
default[:graphite][:http_port]        = 80
default[:graphite][:http_server_name] = "graphite-web"
#Carbon Settings
default[:grahite][:storage_dir]               = "/var/lib/carbon/"
default[:grahite][:local_data_dir]            = "/var/lib/carbon/whisper/"
default[:grahite][:enable_log_rotation]       = true
default[:grahite][:max_updates_per_second]    = 500
default[:grahite][:max_creates_per_miniute]   = 50
default[:grahite][:line_receiver_interface]   = "0.0.0.0"
default[:grahite][:line_receiver_port]        = 2003
default[:grahite][:enable_udp_listener]       = false
default[:grahite][:udp_receiver_interface]    = "0.0.0.0"
default[:grahite][:udp_receiver_port]         = 2003
default[:grahite][:pickle_receiver_interface] = "0.0.0.0"
default[:grahite][:pickle_receiver_port]      = 2004
default[:grahite][:log_listener_connections]  = true
default[:grahite][:use_insecure_unpickler]    = false
default[:grahite][:cache_query_interface]     = "0.0.0.0"
default[:grahite][:cache_query_port]          = 7002
default[:grahite][:use_flow_control]          = true
default[:grahite][:log_updates]               = false
default[:grahite][:log_cache_hits]            = false
default[:grahite][:log_cache_queue_sorts]     = true
default[:grahite][:cache_write_strategy]      = "sorted"
default[:grahite][:whisper_fallocate_create]  = true
default[:grahite][:whisper_autoflush]         = false
```

Usage
-----
#### grahite::default

e.g.
Just include `grahite` in your node's `run_list`:

```json
{
  "name":"my_node",
  "run_list": [
    "recipe[graphite]"
  ]
}
```

License and Authors
-------------------
License: MIT License

Authors: Marcy


CHANGELOG.md
grahite CHANGELOG
=================

This file is used to list changes made in each version of the grahite cookbook.

0.1.0
-----
- [your_name] - Initial release of grahite

- - -
Check the [Markdown Syntax Guide](http://daringfireball.net/projects/markdown/syntax) for help with Markdown.

The [Github Flavored Markdown page](http://github.github.com/github-flavored-markdown/) describes the differences between markdown on github and standard markdown.


test/integration/default/serverspec/defalut_spec.rb
require 'serverspec'
require 'pathname'

include Serverspec::Helper::Exec
include Serverspec::Helper::DetectOS

RSpec.configure do |c|
  c.before :all do
    c.os = backend(Serverspec::Commands::Base).check_os
    c.path = '/sbin:/usr/bin'
  end
end

%w{graphite-web graphite-web-selinux MySQL-python python-carbon}.each do |pack|
  describe package(pack) do
    it { should be_installed }
  end
end

describe service('httpd') do
  it { should be_enabled }
  it { should be_running }
end

describe service('carbon-cache') do
  it { should be_enabled }
  it { should be_running }
end

describe port(80) do
  it { should be_listening }
end

describe port(3306) do
  it { should be_listening }
end

describe file('/etc/graphite-web/local_settings.py') do
  it { should be_file }
  its(:content) { should match /TIME_ZONE\s*=\s*'Asia\/Tokyo'/ }
  its(:content) { should match /'PASSWORD':\s'graphite'/ }
end

describe file('/usr/lib/python2.6/storage') do
  it { should be_directory }
end

describe file('/usr/lib/python2.6/storage/log/webapp') do
  it { should be_directory }
end

describe file('/usr/lib/python2.6/storage/whisper') do
  it { should be_linked_to '/var/lib/carbon/whisper' }
end

describe file('/etc/httpd/conf.d/graphite-web.conf') do
  it { should be_file }
  its(:content) { should match /^<VirtualHost\s\*:80>$/ }
  its(:content) { should match /^\s*ServerName\sgraphite-web$/ }
end

describe file('/etc/carbon/carbon.conf') do
  it { should be_file }
  its(:content) { should match /^STORAGE_DIR\s*=\s*\/var\/lib\/carbon\/$/ }
  its(:content) { should match /^LOCAL_DATA_DIR\s*=\s*\/var\/lib\/carbon\/whisper\/$/ }
  its(:content) { should match /^ENABLE_LOGROTATION\s*=\s*True$/ }
  its(:content) { should match /^MAX_UPDATES_PER_SECOND\s*=\s*500$/ }
  its(:content) { should match /^MAX_CREATES_PER_MINUTE\s*=\s*50$/ }
  its(:content) { should match /^LINE_RECEIVER_INTERFACE\s*=\s*0\.0\.0\.0$/ }
  its(:content) { should match /^LINE_RECEIVER_PORT\s*=\s*2003$/ }
  its(:content) { should match /^ENABLE_UDP_LISTENER\s*=\s*False$/ }
  its(:content) { should match /^UDP_RECEIVER_INTERFACE\s*=\s*0\.0\.0\.0$/ }
  its(:content) { should match /^UDP_RECEIVER_PORT\s*=\s*2003$/ }
  its(:content) { should match /^PICKLE_RECEIVER_INTERFACE\s*=\s*0\.0\.0\.0$/ }
  its(:content) { should match /^PICKLE_RECEIVER_PORT\s*=\s*2004$/ }
  its(:content) { should match /^LOG_LISTENER_CONNECTIONS\s*=\s*True$/ }
  its(:content) { should match /^USE_INSECURE_UNPICKLER\s*=\s*False$/ }
  its(:content) { should match /^CACHE_QUERY_INTERFACE\s*=\s*0\.0\.0\.0$/ }
  its(:content) { should match /^CACHE_QUERY_PORT\s*=\s*7002$/ }
  its(:content) { should match /^USE_FLOW_CONTROL\s*=\s*True$/ }
  its(:content) { should match /^LOG_UPDATES\s*=\s*False$/ }
  its(:content) { should match /^LOG_CACHE_HITS\s*=\s*False$/ }
  its(:content) { should match /^LOG_CACHE_QUEUE_SORTS\s*=\s*True$/ }
  its(:content) { should match /^CACHE_WRITE_STRATEGY\s*=\s*sorted$/ }
  its(:content) { should match /^WHISPER_AUTOFLUSH\s*=\s*False$/ }
  its(:content) { should match /^WHISPER_FALLOCATE_CREATE\s*=\s*True$/ }
end


attributes/default.rb
#Graphite Web Settings
default[:graphite][:timezone]         = "Asia/Tokyo"
default[:graphite][:mysql_password]   = "graphite"
default[:graphite][:http_port]        = 80
default[:graphite][:http_server_name] = "graphite-web"
#Carbon Settings
default[:graphite][:storage_dir]               = "/var/lib/carbon/"
default[:graphite][:local_data_dir]            = "/var/lib/carbon/whisper/"
default[:graphite][:enable_log_rotation]       = true
default[:graphite][:max_updates_per_second]    = 500
default[:graphite][:max_creates_per_miniute]   = 50
default[:graphite][:line_receiver_interface]   = "0.0.0.0"
default[:graphite][:line_receiver_port]        = 2003
default[:graphite][:enable_udp_listener]       = false
default[:graphite][:udp_receiver_interface]    = "0.0.0.0"
default[:graphite][:udp_receiver_port]         = 2003
default[:graphite][:pickle_receiver_interface] = "0.0.0.0"
default[:graphite][:pickle_receiver_port]      = 2004
default[:graphite][:log_listener_connections]  = true
default[:graphite][:use_insecure_unpickler]    = false
default[:graphite][:cache_query_interface]     = "0.0.0.0"
default[:graphite][:cache_query_port]          = 7002
default[:graphite][:use_flow_control]          = true
default[:graphite][:log_updates]               = false
default[:graphite][:log_cache_hits]            = false
default[:graphite][:log_cache_queue_sorts]     = true
default[:graphite][:cache_write_strategy]      = "sorted"
default[:graphite][:whisper_fallocate_create]  = true
default[:graphite][:whisper_autoflush]         = false
